services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/attachment/application/ApplicationAttachmentContentService.kt

```diff
@@ -19,18 +19,26 @@ class ApplicationAttachmentContentService(
     val contentRepository: ApplicationAttachmentContentRepository,
     val fileClient: FileClient,
 ) {
-    fun save(attachmentId: UUID, content: ByteArray) {
-        contentRepository.save(ApplicationAttachmentContentEntity(attachmentId, content))
+    fun save(
+        filename: String,
+        contentType: MediaType,
+        applicationId: Long,
+        content: ByteArray
+    ): String {
+        val blobPath = generateBlobPath(applicationId)
+        fileClient.upload(Container.HAKEMUS_LIITTEET, blobPath, filename, contentType, content)
+        return blobPath
     }
 
-    fun delete(attachment: AttachmentMetadata) =
-        with(attachment) {
-            logger.info { "Deleting attachment $id if blob defined. Blob=$blobLocation." }
-            blobLocation?.let { fileClient.delete(Container.HAKEMUS_LIITTEET, it) }
+    fun delete(blobPath: String?) {
+        if (blobPath == null) {
+            logger.info { "Blob path is null, nothing to delete" }
+            return
         }
+        fileClient.delete(Container.HAKEMUS_LIITTEET, blobPath)
+    }
 
     fun deleteAllForApplication(applicationId: Long) {
-        logger.info { "Deleting all attachments from application $applicationId" }
```

Comment:
Was there something wrong with this? Why was it removed?

Reply:
This is called from only one place in `ApplicationAttachmentService` and there is logging there:
```
    fun deleteAllAttachments(application: Application) {
        attachmentContentService.deleteAllForApplication(application.id!!)
        metadataService.deleteAllAttachments(application.id)
        logger.info { "Deleted all attachments from application ${application.id}" }
    }
```

Reply:
That logging happens after the operation. It's also useful to log before doing something, especially when that something is related to external services. So if there's a connection error or something, the logs have some context on what the service was trying to do when the error happened.

Reply:
On lower lever at `BlobFileClient.deleteAllByPrefix(..)` this is already logged:
```
    override fun deleteAllByPrefix(container: Container, prefix: String) {
        logger.info { "Deleting all Blobs from container $container with prefix $prefix" }
        val containerClient = getContainerClient(container)
        val blobUrls =
            containerClient.listBlobsByHierarchy(prefix).map {
                if (it.isPrefix) {
                    throw UnexpectedSubdirectoryException(container, it.name)
                }
                containerClient.getBlobClient(it.name).blobUrl
            }
        if (blobUrls.isEmpty()) {
            return
        }
        batchClient.deleteBlobs(blobUrls, DeleteSnapshotsOptionType.INCLUDE).forEach {
            logger.info { "Deleted blob ${it.request.url} with status ${it.statusCode}" }
        }
        logger.info { "Deleted all Blobs from container $container with prefix $prefix" }
    }
```

I have it little difficult to pinpoint the best location for logging but I did some adjustments here and  there.